### PR TITLE
Align PrePostProcessor methods exposed in JS API

### PR DESCRIPTION
**Pull request by almilosz**
### Details:
 - Expose basic scope of `InputInfo` `InputModelInfo` `InputTensorInfo` `PreProcessSteps` classes
 - Align exposed PrePostProcessor methods to C++ API
 

### Tickets:
 - *[123742]*

### CODE REVIEW

File: samples/js/node/hello_classification/hello_classification.js [hunk 1]
```diff
@@ -44,12 +44,11 @@ async function main(modelPath, imagePath, deviceName) {
...
   //----------------- Step 4. Apply preprocessing ------------------------------
-  new ov.PrePostProcessor(model)
-    .setInputTensorShape(shape)
-    .preprocessResizeAlgorithm(ov.resizeAlgorithm.RESIZE_LINEAR)
-    .setInputTensorLayout('NHWC')
-    .setInputModelLayout('NCHW')
-    .build();
+  const _ppp = new ov.PrePostProcessor(model);
+  _ppp.input(0).tensor().setShape(shape).setLayout('NHWC');
+  _ppp.input(0).preprocess().resize(ov.resizeAlgorithm.RESIZE_LINEAR);
```
Comment (vishniakov-nikolai) [inline samples/js/node/hello_classification/hello_classification.js:49]: We have to move `resizeAlgorithm` to preprocess property

File: samples/js/node/hello_reshape_ssd/hello_reshape_ssd.js [hunk 2]
```diff
@@ -51,14 +51,17 @@ async function main(modelPath, imagePath, deviceName) {
...
-    .setInputTensorLayout('NHWC')
-    .setInputModelLayout('NCHW')
-    // TODO: add output tensor element type setup
-    .build();
+  const _ppp = new ov.PrePostProcessor(model);
+  _ppp.input(0).preprocess().resize(ov.resizeAlgorithm.RESIZE_LINEAR);
+
+  _ppp.input(0).tensor()
+    .setShape(shape)
+    .setElementType(ov.element.u8)
```
Comment (vishniakov-nikolai) [inline samples/js/node/hello_reshape_ssd/hello_reshape_ssd.js:59]: And `ov.element` to `ov.Type`

File: src/bindings/js/node/src/preprocess/input_info.cpp [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+#include "preprocess/input_model_info.hpp"
+#include "preprocess/input_tensor_info.hpp"
+#include "preprocess/preprocess_steps.hpp"
+
+InputInfo::InputInfo(const Napi::CallbackInfo& info) : Napi::ObjectWrap<InputInfo>(info){};
+
+Napi::Function InputInfo::GetClassConstructor(Napi::Env env) {
+    return DefineClass(env,
+                       "InputInfo",
+                       {InstanceMethod("tensor", &InputInfo::tensor), InstanceMethod("preprocess", &InputInfo::preprocess), InstanceMethod("model", &InputInfo::model)});
```
Comment (vishniakov-nikolai) [inline src/bindings/js/node/src/preprocess/input_info.cpp:16]:
```suggestion
                       {InstanceMethod("tensor", &InputInfo::tensor), 
                        InstanceMethod("preprocess", &InputInfo::preprocess), 
                        InstanceMethod("model", &InputInfo::model)});
```

File: src/bindings/js/node/tests/basic.test.js [hunk 4]
```diff
@@ -14,8 +14,109 @@ const compiledModel = core.compileModelSync(model, 'CPU');
...
+  });
+
+  it('model().setLayout() throws', () => {
+    assert.throws(() => new ov.PrePostProcessor(model).input(0).model().setLayout('NCHW', 'extraArg'),
+      /Wrong number of parameters./);
+  });
+
+  it('model().setLayout() throws', () => {
+    assert.throws(() => new ov.PrePostProcessor(model).input(0).model().setLayout('invalidLayout')
+    );
```
Comment (vishniakov-nikolai) [inline src/bindings/js/node/tests/basic.test.js:113]:
```suggestion
    assert.throws(() => 
      new ov.PrePostProcessor(model).input(0).model().setLayout('invalidLayout'));
```

**Review by vishniakov-nikolai — COMMENTED** [review]
Looks great!
Additionally `ppp.output` needs to be covered

**Comment by vishniakov-nikolai** [comment]
There is no new PrePostProcessor definition in `addon.ts`

File: src/bindings/js/node/src/preprocess/input_info.cpp (hunk 3 above)
Comment (almilosz) [reply src/bindings/js/node/src/preprocess/input_info.cpp:16]: :thumbsup:

File: src/bindings/js/node/tests/basic.test.js (hunk 4 above)
Comment (almilosz) [reply src/bindings/js/node/tests/basic.test.js:113]: 👍

File: src/bindings/js/node/lib/addon.ts [hunk 5]
```diff
@@ -72,14 +72,28 @@ interface Output {
   getPartialShape(): number[];
 }
 
+interface InputTensorInfo {
+    setElementType(elementType: element | string ): InputTensorInfo;
+    setLayout(layout: string): InputTensorInfo;
+    setShape(shape: Int32Array | Array<number>): InputTensorInfo;
```
Comment (vishniakov-nikolai) [inline src/bindings/js/node/lib/addon.ts:78]:
```suggestion
    setShape(shape: number[]): InputTensorInfo;
```

Comment (almilosz) [reply src/bindings/js/node/lib/addon.ts]: 👍

**Review by vishniakov-nikolai — APPROVED** [review]
Great! LGTM.

**Pull request merged**